ovve_ui/utils/comms_link.py

```diff
@@ -0,0 +1,278 @@
+            
+            # DEBUG
+            print ("Packet Written:")
+            print(''.join(r'\x'+hex(letter)[2:] for letter in cmd_byteData))
+            print("Sent back SEQ and CRC: ")
+            print (int.from_bytes(cmd_byteData[0:2], byteorder='little'))
+            print (int.from_bytes(cmd_byteData[20:], byteorder='big'))
+            # END DEBUG
+
+            #Update dict only if there is valid data
+            if validData == True:
+                # any settings set will not retunr correctly yet until Arduino sets set values correctly
+                # We can use the settings values like in simulator if so desired or maybe compare them
+                params_dict['run_state'] = self.settings.run_state
+                params_dict['seq_num'] = in_pkt['sequence_count']
+                params_dict['packet_version'] = in_pkt['packet_version']
+                params_dict['mode'] = in_pkt['mode_value']
+                params_dict['resp_rate_meas'] = in_pkt['respiratory_rate_measured']
+                params_dict['resp_rate_set'] = in_pkt['respiratory_rate_set']
+                params_dict['tv_meas'] = in_pkt['tidal_volume_measured']
+                params_dict['tv_set'] = in_pkt['tidal_volume_set']
+                params_dict['ie_ratio_meas'] = in_pkt['ie_ratio_measured']
+                params_dict['ie_ratio_set'] = in_pkt['ie_ratio_set']
+                params_dict['peep'] = in_pkt['peep_value_measured']
+                params_dict['ppeak'] = in_pkt['peak_pressure_measured']
+                params_dict['pplat'] = in_pkt['plateau_value_measurement']
+                params_dict['pressure'] = in_pkt['pressure_measured']
+                params_dict['flow'] = in_pkt['flow_measured']
+                params_dict['tv_insp'] = in_pkt['volume_in_measured']
+                params_dict['tv_exp'] = in_pkt['volume_out_measured']
+                params_dict['tv_rate'] = in_pkt['volume_rate_measured']
+                # Not sure what is tv_min keeping it like in sumulator for now
+                params_dict['tv_min'] = random.randrange(475, 575)
+                params_dict['control_state'] = 0
+                params_dict['battery_level'] = in_pkt['battery_level']
+                # is this class variable necessary?
+                # self. seqnum = params_dict['seq_num']
+                self.comms_adapter.update_params(params_dict)
+                # we got here
+                # print ("UI parameters updated")
+
+            #
+            #time.sleep(1)
+
+    def init_serial(self):
+    
+        #global ser          #Must be declared in Each Function
+        self.ser = serial.Serial()
+        self.ser.baudrate = self.BAUD
+        self.ser.port = self.PORT 
+        #a little less than polling spped from arduino
+        self.ser.timeout = self.SER_TIMEOUT
+        self.ser.open()          #Opens SerialPort
+
+        # print port open or closed
+        if self.ser.isOpen():
```

Comment:
I think we need error handling and retries here.  What do we do if we lose the serial connection?  Can we enter a recovery mode?